src/cartesian/Bar.tsx

```diff
@@ -146,7 +146,7 @@ export class Bar extends PureComponent<Props, State> {
     offset,
   }: {
     props: Props;
-    item: Bar;
```

Comment:
was `Bar` referencing the class in this case? :o. Is that not still more accurate than a generic ReactElement?

Reply:
Could we do `ReactElement<BarProps>` to give the needed specificity?

Reply:
So two answers, one short term (this PR) one long term (recharts future).

Short term: Yes, using `Bar` is more accurate. But it is only accurate on the receiving end! In the sending end there is no guarantee. `generateCategoricalChart` sends ReactElement. So this was a false positive, going against Barbara Liskov principle.

Long term: In my opinion, architecture where a component requires a _rendered instance of itself_ to look up in the global state before it decides if it can render - is not a good architecture. This whole function should be different and this argument should not exist.

Reply:
Agreed with the long term for sure, one of the major issues with why generateCategoricalChart is how it is. 

Unfortunate that short term we can't have some sort of guarantee of what we're receiving, but I guess that's just the cards we've been deal with this little mess. 

Thanks